src/object/authenticate.c

```diff
@@ -6905,19 +6905,20 @@ char *
 au_get_user_name (MOP obj)
 {
   DB_VALUE value;
-  int error;
-  char *name;
+  db_make_null (&value);
 
-  name = NULL;
-  error = obj_get (obj, "name", &value);
+  int error = obj_get (obj, "name", &value);
   if (error == NO_ERROR)
     {
       if (IS_STRING (&value) && !DB_IS_NULL (&value) && db_get_string (&value) != NULL)
 	{
-	  name = db_get_string (&value);
+	  return ws_copy_string (db_get_string (&value));
```

Comment:
when returned, original value is still not freed :)
```c
	  name = ws_copy_string (db_get_string (&value));

...
  db_value_clear (&value);

  return name;
```

Same for the other case.